ca/ca.go

```diff
@@ -396,9 +402,17 @@ func (ca *CAImpl) CompleteOrder(order *core.Order) {
 		authz.RUnlock()
 	}
 
+	// Build a list of approved extensions to include in the certificate
+	var extensions []pkix.Extension
+	for _, ext := range order.ParsedCSR.Extensions {
+		if isOCSPMustStapleExtension(ext) {
+			extensions = append(extensions, ext)
```

Comment:
Directly appending the extension from the CSR is still _slightly_ dangerous, as we are directly carrying over the critical bit. Boulder has a [pre-built OCSP must-staple extension](https://github.com/letsencrypt/boulder/blob/34b705f29348930ab6f6ac3c8b70f3dcdd061e24/issuance/cert.go#L202), and it uses that both to [detect must-staple in the CSR](https://github.com/letsencrypt/boulder/blob/34b705f29348930ab6f6ac3c8b70f3dcdd061e24/issuance/cert.go#L348) and [include must-staple in the resulting cert](https://github.com/letsencrypt/boulder/blob/34b705f29348930ab6f6ac3c8b70f3dcdd061e24/issuance/cert.go#L300). That way Boulder can guarantee that the extension is never critical.

What you have here is probably fine for Pebble, but if you want to be even safer that's a pattern you could follow.

Reply:
ah nice, I'll do that! I wasn't sure how risky the critical value was, but that makes sense.